lib/pavilion/builder.py

```diff
@@ -608,6 +612,16 @@ def _setup_build_dir(self, dest):
                     "Could not copy extra file '{}' to dest '{}': {}"
                     .format(path, dest, err))
 
+    def make_files(self, files, dst):
+        for file in files:
+            # Assumes that each file is a path, e.g., 'somedir/file', or './file'
+            dirname = os.path.dirname(file)
```

Comment:
What happens if the file already exists? 
It should probably only be an error if you're trying to overwrite a directory with a regular file. 
All the cases still have to be handled though.